### PR TITLE
[FrameworkBundle][SodiumVault] Create secrets directory only when it is used

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

FWB `secrets` is enabled by default. After upgrading an app to 4.4, a directory is created (and checked on every request) even if I don't use this feature. Can't we just disable it by default btw?

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Secrets/SodiumVault.php [hunk 1]
```diff
@@ -150,6 +171,8 @@ public function remove(string $name): bool
 
     public function list(bool $reveal = false): array
     {
+        $this->init();
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Secrets/SodiumVault.php]: this won't have the effect you want to achieve.
instead, the mkdir should be move inside loadKeys and read-only method should be able to work when the dir is missing

Comment (fancyweb) [reply src/Symfony/Bundle/FrameworkBundle/Secrets/SodiumVault.php]: It works but it's not optimal for sure. I'll refacto the whole class then to only create the dir when it is actually needed.

Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/Secrets/SodiumVault.php]: it works for you (c) but as soon as you'll try accessing a missing env var, the directory will be created - so it doesn't work for everyone :)

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @fancyweb.

**Comment by phcorp** [comment]
Thank you, symfony recipe should also be updated: https://github.com/symfony/recipes/tree/master/symfony/framework-bundle/4.4/config

**Comment by nicolas-grekas** [comment]
See https://github.com/symfony/recipes/pull/708 about the recipe.

**Pull request merged**